crates/rocksdb/src/db_manager.rs

```diff
@@ -460,4 +464,24 @@ impl DbWatchdog {
     }
 }
 
+fn check_memory_limit(opts: &CommonOptions) {
+    if let Some(process_memory_size) = opts.process_total_memory_size() {
+        let memory_ratio =
+            opts.rocksdb_total_memory_size.get() as f64 / process_memory_size.get() as f64;
+        if memory_ratio < 0.5 {
+            warn!(
+                "'rocksdb-total-memory-size' parameter is set to {}, less than half the process memory limit of {}. Roughly 75% of process memory should be given to RocksDB",
+                ByteCount::from(opts.rocksdb_total_memory_size),
+                ByteCount::from(process_memory_size),
+            )
+        } else if memory_ratio > 0.9 {
+            error!(
+                "'rocksdb-total-memory-size' parameter is set to {}, more than 90% of the process memory limit of {}. This risks an OOM under load; roughly 75% of process memory should be given to RocksDB",
+                ByteCount::from(opts.rocksdb_total_memory_size),
+                ByteCount::from(process_memory_size),
+            )
```

Comment:
An important case that's worth covering is if the ratio is > 1.0. For instance if users are running default restate's config (6GB) on a pod with 3GB memory limit. I understand that this will be covered by the >0.9 case but better make it explicit.